clusterloader2/pkg/chaos/nodes.go

```diff
@@ -98,11 +99,15 @@ func (k *NodeKiller) pickNodes() ([]v1.Node, error) {
 	rand.Shuffle(len(nodes), func(i, j int) {
 		nodes[i], nodes[j] = nodes[j], nodes[i]
 	})
-	numNodes := int(k.config.FailureRate * float64(len(nodes)))
-	if len(nodes) > numNodes {
-		return nodes[:numNodes], nil
+	expectedNodesToFail := k.config.FailureRate * float64(len(nodes))
+	nodesToFail := nodes[:0]
+	klog.Infof("%s: %d nodes available, expecting to fail %f nodes", k, len(nodes), expectedNodesToFail)
+	for _, node := range nodes {
+		if rand.Float64() < k.config.FailureRate {
```

Comment:
This is already randomized, we call rand.Shuffle in line 99

Reply:
They are randomized, but the size is constant.

And if we are expecting that less than a single node will fail (as for load test, with failure rate 0.01, and less than 100 eligible nodes, as some of them are running Prometheus), no node is scheduled.

Reply:
I see, good debugging!
But, wouldn't be better to simply set numNodes to 1 if it's 0?
What I don't like about your solution is that, while it'll kill the `FailureRate` of nodes on average, we'll have test-runs where no nodes were killed and we'll have runs where 1+ nodes were killed. It's quite likely that it will make the test flaky - most likely it won't be failing but it can result in big variance of many metrics.

Reply:
Sure, changed that to get more stable tests. Also, increased minimal interval in load test to prevent overkilling.